### PR TITLE
Validate input

### PR DESCRIPTION
**Pull request by thedmail**

### CODE REVIEW

File: compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt [hunk 1]
```diff
@@ -839,3 +840,55 @@ private val firaSansFamily = FontFamily()
...
+        }
+    }
+
+    @Composable
+    fun ValidatingInputTextField(
+        email: String,
+        updateState: (String) -> Unit,
+        validatorHasErrors: Boolean
+    ) {
+        val emailViewModel = EmailViewModel()
```
Comment (riggaroo) [inline compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt]: This should use the recommended pattern here: https://developer.android.com/develop/ui/compose/libraries#viewmodel

i.e. import ` implementation 'androidx.lifecycle:lifecycle-viewmodel-compose:2.8.5'`
and then `viewModel: MyViewModel = viewModel()`

**Review by riggaroo — CHANGES_REQUESTED** [review]
Update viewmodel usage

File: compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt (hunk 1 above)
Comment (thedmail) [reply compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt]: Doesn't work when I do that. If I try it in the function signature, I get an "expecting an element" error. (There's nothing to pass, which is why we're declaring the variable locally.)

But if I try declaring it outside the signature, as a local variable, based on the pattern in the doc:

val emailViewModel: EmailViewModel=emailViewModel()

I get an error that emailViewModel is undeclared.

Comment (riggaroo) [reply compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt]: fixed it for you

**Review by riggaroo — APPROVED** [review]

File: compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt (hunk 1 above)
Comment (thedmail) [reply compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt]: ty!

**Pull request merged**